src/main/groovy/com/github/j2objccontrib/j2objcgradle/DependencyResolver.groovy

```
@@ -42,20 +44,50 @@ class DependencyResolver {
     }
 
     void configureAll() {
+        project.configurations.getByName('j2objcTranslationClosure').each { File it ->
+            // These are the resolved files, NOT the dependencies themselves.
+            visitTranslationClosureFile(it)
+        }
         project.configurations.getByName('j2objcTranslation').each { File it ->
             // These are the resolved files, NOT the dependencies themselves.
-            visitTranslateFile(it)
+            visitTranslationLibrary(it)
         }
         project.configurations.getByName('j2objcLinkage').dependencies.each {
             visitLink(it)
         }
     }
 
-    protected void visitTranslateFile(File depFile) {
+    protected void visitTranslationClosureFile(File depFile) {
         j2objcConfig.translateSourcepaths(depFile.absolutePath)
         j2objcConfig.enableBuildClosure()
     }
 
+    /**
+     * Adds to the main java sourceSet a to-be-generated directory that contains the contents
+     * of `j2objcTranslation` dependency libraries (if any).
+     */
+    static void configureSourceSets(Project project) {
+        JavaPluginConvention javaConvention = project.getConvention().getPlugin(JavaPluginConvention)
+        SourceSet sourceSet = javaConvention.sourceSets.findByName(SourceSet.MAIN_SOURCE_SET_NAME)
+        sourceSet.java.srcDirs(project.file("${project.buildDir}/translationExtraction"))
+        Copy copy = project.tasks.create('j2objcTranslatedLibraryExtraction', Copy,
+                {Copy task -> task.into(project.file("${project.buildDir}/translationExtraction"))})
+        project.tasks.getByName(sourceSet.compileJavaTaskName).dependsOn(copy)
+    }
+
+    protected void visitTranslationLibrary(File depFile) {
```

1) done
2) done
3) done
4) the idea is you would generally only convert one library per project using j2objcTranslation.  anyway, I added L74 to fail fast on duplicate entries.